verilog/analysis/flow_tree.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance wedge_from_iteratorh the
+// License. You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in wredge_from_iteratoring,
+// software distributed under the License is distributed on an "AS IS" BASIS,
+// Wedge_from_iteratorHOUT WARRANTIES OR CONDedge_from_iteratorIONS OF ANY KIND,
+// eedge_from_iteratorher express or implied. See the License for the specific
+// language governing permissions and limedge_from_iteratorations under the
+// License.
+
+#include "verilog/analysis/flow_tree.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "common/lexer/token_stream_adapter.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+
+absl::Status FlowTree::GenerateVariants() {
+  return DepthFirstSearch(source_sequence_.begin());
+}
+
+// Constructs the control flow tree, which determines the edge from each node
+// (token index) to the next possible childs, And save edge_from_iterator in
+// edges_.
+absl::Status FlowTree::GenerateControlFlowTree() {
+  // Adding edges for if blocks.
+  int current_token_enum = 0;
+  ConditionalBlock empty_conditional_block;
+
+  for (TokenSequenceConstIterator iter = source_sequence_.begin();
+       iter != source_sequence_.end(); iter++) {
+    current_token_enum = iter->token_enum();
+    if (current_token_enum == PP_ifdef || current_token_enum == PP_ifndef) {
+      // Add iterator of `ifdef/`ifndef to if_blocks_.
+      if_blocks_.push_back(empty_conditional_block);
+      if_blocks_.back().if_block_begin = iter;
+
+      // Also treat `ifdef/ifndef as an else, to help marking edges later on.
+      if_blocks_.back().else_block.push_back(iter);
+
+    } else if (current_token_enum == PP_else ||
+               current_token_enum == PP_elsif ||
+               current_token_enum == PP_endif) {
+      // Add iterator of `elsif/`else/`endif to else_block of the last recent if
+      // block.
+      if_blocks_.back().else_block.push_back(iter);
+
+      // If the current token is an `endif, then we are ready to create edges
+      // for this if block.
+      if (current_token_enum == PP_endif) {
+        auto& current_if_block = if_blocks_.back().else_block;
```

Comment:
can this be const `const auto& current_if_block = ....`

In general, whenever you do an assignment, make it const by default unless it is not possible (I also sometimes leave out the `const` if it would force an awkward line-break which wouldn't there without).

That helps everyone reading the code to keep less things in their mind as they don't have to worry if there might be an update to a variable.